### PR TITLE
fix: stop at nakamoto block height

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes https://github.com/stacks-network/sbtc/issues/778 and addresses part of https://github.com/stacks-network/sbtc/issues/685. 

## Changes

* Change the `get_tenure` return type to include the anchor block hash and anchor block height.
* Make sure `fetch_unknown_ancestors` stops at the nakamoto start height.

## Testing Information

I haven't updated the tests, so this will fail.

## Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: signer/src/stacks/api.rs [hunk 1]
```diff
@@ -860,41 +901,41 @@ pub async fn fetch_unknown_ancestors<S, D>(
...
-                last_chain_length = %block.header.chain_length,
+                last_chain_length = %tenure.anchor_block_height,
                 "Stopping, since we have fetched all Nakamoto blocks"
             );
             break;
         }
+        // Tenure blocks are always non-empty, so no need to worr about the
+        // early break.
+        let Some(block) = tenure.blocks.last() else {
+            break;
```
Comment (matteojug) [inline signer/src/stacks/api.rs:929]: Should we log/raise here? Just to be sure we can check if something that shouldn't happen actually happens.

**Review by matteojug — COMMENTED** [review]
NIce! So far LGTM

File: signer/src/stacks/api.rs (hunk 1 above)
Comment (djordon) [reply signer/src/stacks/api.rs:929]: I'm going to maintain this invariant in the struct.

File: signer/src/stacks/api.rs [hunk 2]
```diff
@@ -860,41 +901,41 @@ pub async fn fetch_unknown_ancestors<S, D>(
...
+        if tenure.anchor_block_height <= nakamoto_start_height {
             tracing::info!(
                 %nakamoto_start_height,
-                last_chain_length = %block.header.chain_length,
+                last_chain_length = %tenure.anchor_block_height,
                 "Stopping, since we have fetched all Nakamoto blocks"
             );
             break;
         }
+        // Tenure blocks are always non-empty, so no need to worr about the
```
[the remark this answers is not in the document]
Comment (djordon) [reply signer/src/stacks/api.rs]: Fixed in https://github.com/stacks-network/sbtc/pull/781/commits/81a91336b13335ddc715c3972b5795c60346608e

File: signer/src/stacks/api.rs [hunk 3]
```diff
@@ -241,7 +241,37 @@ pub struct TenureBlocks {
...
+            anchor_block_hash: info.burn_block_hash.into(),
+            anchor_block_height: info.burn_block_height,
+        })
+    }
+
+    /// Get all the blocks contained in this object.
+    ///
+    /// # Note
+    ///
+    /// The struct need not contain all the blocks in a tenure.
```
Comment (matteojug) [inline signer/src/stacks/api.rs:260]: nit? also below
```suggestion
    /// The struct doesn't need to contain all the blocks in a tenure.
```

File: signer/tests/integration/transaction_coordinator.rs [hunk 4]
```diff
@@ -1112,8 +1111,14 @@ async fn sign_bitcoin_transaction() {
         signers.push((ctx, db, kp, network));
     }
 
+    let (broadcast_stacks_tx, _rx) = tokio::sync::broadcast::channel(1);
+
+    let mut stacks_tx_receiver = broadcast_stacks_tx.subscribe();
+    let stacks_tx_receiver_task = tokio::spawn(async move { stacks_tx_receiver.recv().await });
+
     // 3. Check that there are no DKG shares in the database.
```
Comment (matteojug) [inline signer/tests/integration/transaction_coordinator.rs:1119]: nit: wrong comment?

**Review by matteojug — APPROVED** [review]

File: signer/tests/integration/transaction_coordinator.rs (hunk 4 above)
Comment (djordon) [reply signer/tests/integration/transaction_coordinator.rs]: Fixed in https://github.com/stacks-network/sbtc/pull/781/commits/2132edc0310007a2d39b6efa0ac302fa6c1e29e6

File: signer/src/stacks/api.rs (hunk 3 above)
Comment (djordon) [reply signer/src/stacks/api.rs]: 🙏🏽 updated in https://github.com/stacks-network/sbtc/pull/781/commits/2132edc0310007a2d39b6efa0ac302fa6c1e29e6.

**Pull request merged**